AnimalSorting/AnimalSorting/Animals.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AnimalSorting
+{
+    public class Animals : IEnumerable<Animal>
+    {
+        private Animal[] animals;
+
+        public Animals(Animal[] animals)
+        {
+            if (animals == null)
+                throw new ArgumentNullException("animals array can't be null");
+
+            this.animals = animals;
+        }
+        public IEnumerator<Animal> GetEnumerator()
+        {
+            return new Enumerator(this.animals);
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return this.GetEnumerator();
+        }
+
+        class Enumerator : IEnumerator<Animal>
+        {
+            private int position;
+            private Animal[] animals;
+
+            public Animal Current { get => animals[position]; }
+
+            object IEnumerator.Current => this.Current;
+
+            public Enumerator(Animal[] animals)
+            {
+                this.position = -1;
+                this.animals = animals;
+            }
+
+            public void Dispose()
+            {
+                this.Reset();
+            }
+
+            public bool MoveNext()
+            {
+                position++;
+                if (position >= this.animals.Length)
```

Comment:
You can just return position >= this.animals.Length

Reply:
Done 6824b0a3d4a7fb13b8402881142acfe9fba98794